adding assignment (#1)
tjperry07/hashicorp_writing_exercise: assignment.md

### What is the difference between push, pull, and fetch?

- `git push` - sent changes from a local branch to a remote repo
- `git fetch` - get changes from a remote repo into your tracking branch
- `git pull` - will get changes from a remote branch into your tracking branch and merge them into a local branch

> Often `git push` and `git pull` are described as equivalent. This isn't entirely correct, since under the hood `git pull` does two things. `git push` takes our current branch, and checks to see whether or not there is a tracking branch for a remote repository connected to it. If so, our changes are taken from our branch and pushed to the remote branch. This is how code is shared with a remote repository, you can think of it as "make the remote branch resemble my local branch". This will fail if the remote branch has diverged from your local branch: if not all the commits in the remote branch are in your local branch. When this happens, your local branch needs to be synchronized with the remote branch with git pull or git fetch and git merge.`git fetch` again takes our current branch, and checks to see if there is a tracking branch. If so, it looks for changes in the remote branch, and pulls them into the tracking branch. It does not change your local branch. To do that, you'll need to do `git merge origin/master` (for the "master" branch) to merge those changes into your branch - probably also called "master".`git pull` simply does a `git fetch` followed immediately by `git merge`. This is often what we desire to do, but some people prefer to use git fetch followed by git merge to make sure they understand the changes they are merging into their branch before the merge happens.

Author: ```suggestion
**Often** `git push` and `git pull` are described as equivalent. **This isn't entirely correct**, since **under the hood** `git pull` does two things. `git push` takes our current branch, and checks to see whether or not there is a tracking branch for a remote repository connected to it. **If so**, **our (try and use you for the user)** changes are taken from **our** branch and pushed to the remote branch. This is how code is shared with a remote repository, you can think of it as "make the remote branch resemble my local branch". This will fail if the remote branch has diverged from your local branch. If not all the commits in the remote branch are in your local branch. When this happens, your local branch needs to be synchronized with the remote branch with git pull or git fetch and git merge.`git fetch` **again** takes our current branch, and checks to see if there is a tracking branch. **If so,** it looks for changes in the remote branch, and pulls them into the tracking branch. It does not change your local branch. **To do that,** you'll need to do `git merge origin/**master (using main is more inclusive)**` (for the "master" branch) to merge those changes into your branch - **probably** also called "master".`git pull` simply does a `git fetch` followed immediately by `git merge`. **This is often what we desire to do**, but some people prefer to use git fetch followed by git merge to make sure they understand the changes they are merging into their branch before the merge happens.
```

---

> # What Is the Difference Between Push, Pull, and Fetch?

Author: Changed to sentence case

> - `git push` - Send changes from a local branch to a remote repository

Author: The idea of having a quick summary was great. I capitalized the beginning of the definition to make it easier to see

- `git fetch` - Get changes from a remote repo into your tracking branch
- `git pull` - Gets changes from a remote branch into your tracking branch and merges them into a local branch

> ## Git Push

Author: I broke each command into a section with a heading. This way it's easier to find when searching and easier for users to scan the page.

> `git push`

Author: This is 100% a preference and not required. When explaining commands I add the command at the beginning. Feel free to leave this out.

> Git push shares your local code with a remote repository. Git push works by taking your current branch and checking if there is a tracking branch for a remote repository connected to it. If a tracking branch is found, your changes are taken from your branch and pushed to the remote branch.

Author: In the paragraphs I've made sure to keep shorter sentences, remove jargon, and extra words.

> ### What is the difference between push, pull, and fetch?

Author: ```suggestion
# What Is the Difference Between Push, Pull, and Fetch?
```

> - `git push` - sent changes from a local branch to a remote repo
> - `git fetch` - get changes from a remote repo into your tracking branch
> - `git pull` - will get changes from a remote branch into your tracking branch and merge them into a local branch

Author: ```suggestion
- `git push` - Send changes from a local branch to a remote repository
- `git fetch` - Get changes from a remote repo into your tracking branch
- `git pull` - Gets changes from a remote branch into your tracking branch and merges them into a local branch
```

> Often `git push` and `git pull` are described as equivalent. This isn't entirely correct, since under the hood `git pull` does two things. `git push` takes our current branch, and checks to see whether or not there is a tracking branch for a remote repository connected to it. If so, our changes are taken from our branch and pushed to the remote branch. This is how code is shared with a remote repository, you can think of it as "make the remote branch resemble my local branch". This will fail if the remote branch has diverged from your local branch: if not all the commits in the remote branch are in your local branch. When this happens, your local branch needs to be synchronized with the remote branch with git pull or git fetch and git merge.`git fetch` again takes our current branch, and checks to see if there is a tracking branch. If so, it looks for changes in the remote branch, and pulls them into the tracking branch. It does not change your local branch. To do that, you'll need to do `git merge origin/master` (for the "master" branch) to merge those changes into your branch - probably also called "master".`git pull` simply does a `git fetch` followed immediately by `git merge`. This is often what we desire to do, but some people prefer to use git fetch followed by git merge to make sure they understand the changes they are merging into their branch before the merge happens.

Author: I would break up the paragraph to make it easier to read. 
Remember to cut out extra words. For example `This isn't entirely correct`. 
Try to avoid jargon like `under the hood`.  In the suggestion I made some of this language as an example.
